src/overlays/actors/ovl_Oceff_Storm/z_oceff_storm.c

```diff
@@ -33,22 +31,183 @@ const ActorInit Oceff_Storm_InitVars = {
+    return phi_v1;
+}
+
+void OceffStorm_Init(Actor* thisx, PlayState* play) {
+    s32 pad[2];
+    Player* player = GET_PLAYER(play);
+    OceffStorm* this = THIS;
+
+    OceffStorm_SetupAction(this, OceffStorm_DefaultAction);
+
+    this->posYOffAdd = 0;
+    this->counter = 0;
+    this->primColorAlpha = 0;
+    this->vtxAlpha = 0;
+    this->actor.scale.y = 0.0f;
+    this->actor.scale.z = 0.0f;
+    this->actor.scale.x = 0.0f;
+    this->posYOff = this->posYOffAdd;
+
+    if (this->actor.params == 1) {
+        OceffStorm_SetupAction(this, func_80981B48);
+        this->actor.draw = OceffStorm_Draw2;
+    } else {
+        this->actor.world.pos.y = player->actor.world.pos.y;
+        this->actor.world.pos.x = player->bodyPartsPos[0].x;
+        this->actor.world.pos.z = player->bodyPartsPos[0].z;
+        gSaveContext.jinxTimer = 0;
+        if ((play->interfaceCtx.restrictions.songOfStorms == 0) && !func_8098176C(play)) {
+            Actor_Spawn(&play->actorCtx, play, ACTOR_EN_OKARINA_EFFECT, this->actor.world.pos.x,
+                        this->actor.world.pos.y - 30.0f, this->actor.world.pos.z, 0, 0, 0, 1);
+        }
+    }
+}
+
+void OceffStorm_Destroy(Actor* thisx, PlayState* play) {
+    OceffStorm* this = THIS;
+
+    Magic_Reset(play);
+}
+
+void OceffStorm_DefaultAction(OceffStorm* this, PlayState* play) {
+    f32 cylinderScale;
+
+    switch (gSaveContext.save.playerForm) {
+        default:
+            cylinderScale = 1.0f;
+            break;
+        case PLAYER_FORM_DEKU:
+            cylinderScale = 1.3f;
+            break;
+        case PLAYER_FORM_ZORA:
+            cylinderScale = 1.2f;
+            break;
+        case PLAYER_FORM_GORON:
+            cylinderScale = 2.0f;
+            break;
+    }
+
+    if (this->counter < 10) {
+        this->primColorAlpha = (s8)(this->counter * 10.0f);
+    } else if (this->counter > 60) {
+        this->primColorAlpha = (s8)((70 - this->counter) * 10.0f);
+    } else {
+        this->primColorAlpha = 100;
+    }
+    if ((this->counter < 10) || (this->counter >= 60)) {
+        this->vtxAlpha = 0;
+    } else if (this->counter <= 40) {
+        if (this->vtxAlpha <= 200) {
+            this->vtxAlpha += 10;
+        }
+        this->actor.scale.x = this->actor.scale.z = 0.4f * cylinderScale;
+        this->actor.scale.y = 0.3f;
+    } else if (this->counter > 40) {
+        this->vtxAlpha = (60 - this->counter) * 10;
+    } else {
+        this->actor.scale.x = this->actor.scale.z = 0.4f * cylinderScale;
+        this->vtxAlpha = -1;
+    }
+    if (this->counter > 40) {
+        this->actor.world.pos.y += this->posYOff * 0.01f;
+        this->posYOff += this->posYOffAdd;
+        this->posYOffAdd += 10;
+    }
+    if (this->counter < 70) {
+        this->counter++;
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80981B48(OceffStorm* this, PlayState* play) {
+    if (this->primColorAlpha < 100) {
+        this->primColorAlpha += 5;
+    }
```

Comment:
The lack of Actor_MarkForDeath is still a bug, it will just exist forever. Can use the OoT comment.